pycaret/tests/test_preprocess.py

```diff
@@ -104,6 +104,38 @@ def test_simple_imputer():
     assert result.loc[0, "missing_num_col"] == 100
 
 
+def test_complete_sklearn_pipeline():
+    """
+    Test if the simple imputer in pycaret works with sklearn's pipeline
+    """
+
+    # Load an example dataset and set the features and target
+    data = pycaret.datasets.get_data("juice")
+    target = "Purchase"
+    features = data.columns.tolist()
+    features.remove(target)
+    data_features = data[features]
+    data_target = data[target]
+
+    # Initiate a pycaret pipeline
+    pycaret_preprocessor = pycaret.internal.preprocess.Preprocess_Path_One_Sklearn(
+        train_data=data, target_variable=target, display_types=False
+    )
+
+    # Append classifier to preprocessing pipeline. Now we have a full prediction pipeline.
+    clf = Pipeline(steps=[('preprocessor', pycaret_preprocessor),
+                          ('classifier', xgb.XGBClassifier())],
```

Comment:
Can we use something from sklearn instead? Ideally, we would have our own mock classes for all of this, though (though IIRC sklearn provides something like that, doesn't it?)

Reply:
You mean like using a classifier from sklearn instead of XGBoost? Yes I can do that. I think as  long as the implementation supports the sklearn interface, we can use them in a sklearn pipeline.

Reply:
Yeah, I feel that using something like RandomForest will be a better idea, since XGboost may lag behind any API changes

Reply:
Thanks for the comment! I have changed XGBoost to random forest.